src/easylink/runner.py

```diff
@@ -123,7 +124,123 @@ def main(
     argv.extend(environment_args)
     logger.info(f"Running Snakemake")
     logger.debug(f"Snakemake arguments: {argv}")
-    snake_main(argv)
+
+    # Run snakemake
+    if debug:
+        snake_main(argv)
+    else:
+        _run_snakemake_with_filtered_output(argv, Path(results_dir))
+
+
+def _run_snakemake_with_filtered_output(argv: list[str], results_dir: Path) -> None:
+    """Runs Snakemake with simplified log filtering.
+
+    Parameters
+    ----------
+    argv
+        Snakemake command line arguments.
+    results_dir
+        Directory to save the full Snakemake log.
+    """
+    snakemake_log_file = results_dir / "pipeline.log"
+
+    # Create a filtering output handler that processes lines in real-time
+    class FilteringOutput:
+        """Handles real-time filtering and logging of Snakemake output.
+
+        This class writes all snakemake output to a log file and selectively logs
+        filtered lines to the logger for user visibility.
+
+        Parameters
+        ----------
+        log_file_path
+            The path to the log file where all output will be written.
+
+        """
+
+        def __init__(self, log_file_path: Path):
+            self.log_file = open(log_file_path, "w")
+            self.buffer = ""
+
+        def write(self, text: str) -> int:
+            # Write to log file
+            self.log_file.write(text)
+            self.log_file.flush()
+
+            # Process and log filtered output
+            self.buffer += text
+            while "\n" in self.buffer:
+                line, self.buffer = self.buffer.split("\n", 1)
+                if line.strip():
+                    filtered_line = _filter_snakemake_output_simple(line.strip())
+                    if filtered_line:
+                        logger.info(filtered_line)
+
+            return len(text)
+
+        def flush(self):
+            self.log_file.flush()
+
+        def close(self):
+            # Process and log any remaining buffer content
```

Comment:
Comment: because we remove everything before a `\n` character each time we add to buffer, buffer is guaranteed not to have a `\n` in it, i.e. be a single line.